dedupe/api.py

```diff
@@ -861,7 +859,17 @@ class Dedupe(DedupeMatching, ActiveMatching) :
     """
     canopies = True
 
-    def sample(self, data, sample_size=15000, 
+    def __init__(self, *args, **kwargs):
+        super(Dedupe, self).__init__(*args, **kwargs)
+        data_sample = kwargs.get('data_sample')
+        if data_sample:
+            recs = itertools.chain.from_iterable(data_sample)
```

Comment:
How about having most of this code happen in the ActiveMatching __init__ and then having methods in Dedupe and RecordLink like `_sampledRecords`